### PR TITLE
feat(remove lodash): issue 820

### PR DESCRIPTION
**Pull request by domin191013**
- (open-payments): rotate token
[Remove lodash dependency from backend, mockprovider packages](https://github.com/interledger/rafiki/issues/820)

## Context
Fixed issue #820

## Checklist
- [ ] Related issues linked using `fixes #number`
- [ ] Tests added/updated
- [ ] Documentation added
- [ ] Make sure that all checks pass

### CODE REVIEW

File: packages/backend/src/connector/ilp-routing/lib/prefix-map.ts [hunk 1]
```diff
@@ -64,19 +62,23 @@ export default class PrefixMap<T> {
   *getKeysStartingWith(prefix: string): IterableIterator<string> {
     // TODO: This could be done *much* more efficiently
     const predicate = (key: string): boolean => key.startsWith(prefix)
-    let index = -1
+    // let index = -1
```
Comment (wilsonianb) [inline packages/backend/src/connector/ilp-routing/lib/prefix-map.ts:65]:
```suggestion
```

File: packages/backend/src/connector/ilp-routing/lib/prefix-map.ts [hunk 2]
```diff
@@ -64,19 +62,23 @@ export default class PrefixMap<T> {
...
+      if (predicate(this.prefixes[index])) {
+        yield this.prefixes[index]
+      }
     }
   }
 
   *getKeysPrefixesOf(search: string): IterableIterator<string> {
     const predicate = (key: string): boolean => search.startsWith(key + '.')
-    let index = -1
+    // let index = -1
```
Comment (wilsonianb) [inline packages/backend/src/connector/ilp-routing/lib/prefix-map.ts:76]:
```suggestion
```

File: packages/backend/package.json [hunk 3]
```diff
@@ -75,7 +74,6 @@
     "knex": "^0.95",
     "koa": "^2.14.1",
     "koa-bodyparser": "^4.3.0",
-    "lodash": "^4.17.21",
```
Comment (wilsonianb) [inline packages/backend/package.json:78]: You'll need to update the pnpm lockfile by running
```
pnpm --filter backend remove lodash
pnpm --filter backend remove @types/lodash
```
Ditto for mock-account-provider

File: packages/backend/src/connector/ilp-routing/lib/prefix-map.ts [hunk 4]
```diff
@@ -64,19 +62,21 @@ export default class PrefixMap<T> {
...
+      if (predicate(this.prefixes[index])) {
+        yield this.prefixes[index]
+      }
     }
   }
 
   *getKeysPrefixesOf(search: string): IterableIterator<string> {
     const predicate = (key: string): boolean => search.startsWith(key + '.')
-    let index = -1
     // tslint:disable-next-line:no-conditional-assignment
```
Comment (wilsonianb) [inline packages/backend/src/connector/ilp-routing/lib/prefix-map.ts:75]:
```suggestion
```

File: packages/backend/src/connector/ilp-routing/lib/prefix-map.ts [hunk 5]
```diff
@@ -64,19 +62,21 @@ export default class PrefixMap<T> {
   *getKeysStartingWith(prefix: string): IterableIterator<string> {
     // TODO: This could be done *much* more efficiently
     const predicate = (key: string): boolean => key.startsWith(prefix)
-    let index = -1
     // tslint:disable-next-line:no-conditional-assignment
```
Comment (wilsonianb) [inline packages/backend/src/connector/ilp-routing/lib/prefix-map.ts:65]:
```suggestion
```

File: package.json [hunk 6]
```diff
@@ -30,17 +30,17 @@
...
     "eslint-plugin-react": "^7.31.11",
     "husky": "^8.0.2",
-    "jest": "^28.1",
+    "jest": "^28.1.3",
     "prettier": "^2.8.1",
-    "ts-jest": "^28.0",
+    "ts-jest": "^28.0.8",
     "ts-node-dev": "^1.1.8",
-    "typescript": "^4.2.4"
+    "typescript": "^4.9.4"
```
[the remark this answers is not in the document]
Comment (mkurapov) [reply package.json:43]: @domin191013 as mentioned in DM, let's just keep the `lodash` package removal

[the remark this answers is not in the document]
Comment (domin191013) [reply package.json]: I've manually changed version back to old number, and pushed again.

**Comment by domin191013** [comment]
On main branch, 'pnpm format' is failing.
And because of that my PR also fails in 'pnpm format', but it's  because there's some issues in main branch I guess.

**Review by mkurapov — APPROVED** [review]

**Pull request merged**